src/overlays/actors/ovl_En_Lift_Nuts/z_en_lift_nuts.c

```diff
@@ -28,111 +62,930 @@ const ActorInit En_Lift_Nuts_InitVars = {
+            (CURRENT_DAY == 3) && !(gSaveContext.save.weekEventReg[14] & 0x80)) {
+            gSaveContext.save.weekEventReg[14] |= 0x80;
+        }
+        func_80AEB684(this);
+    } else if (this->textId == 0x27F4 && !(gSaveContext.save.weekEventReg[14] & 0x80)) {
+        Actor_PickUp(&this->actor, play, GI_HEART_PIECE, 500.0f, 100.0f);
+    } else {
+        Actor_PickUp(&this->actor, play, GI_RUPEE_PURPLE, 500.0f, 100.0f);
+    }
+}
+
+void func_80AEB684(EnLiftNuts* this) {
+    this->actionFunc = func_80AEB698;
+}
+
+void func_80AEB698(EnLiftNuts* this, PlayState* play) {
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        if ((gSaveContext.save.weekEventReg[14] & 0x10) && (gSaveContext.save.weekEventReg[14] & 0x20) &&
+            CURRENT_DAY == 3) {
+            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 13);
+            Message_StartTextbox(play, 0x27F5, &this->actor);
+            this->textId = 0x27F5;
+        } else if (func_80AE9B8C() > 0) {
+            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 1);
+            Message_StartTextbox(play, 0x27F0, &this->actor);
+            this->textId = 0x27F0;
+        } else {
+            Message_StartTextbox(play, 0x27F1, &this->actor);
+            this->textId = 0x27F1;
+        }
+        this->actor.flags &= ~ACTOR_FLAG_10000;
+        func_80AEABF0(this);
+        switch (CURRENT_DAY) {
+            case 1:
+                gSaveContext.save.weekEventReg[14] |= 0x10;
+                break;
+            case 2:
+                gSaveContext.save.weekEventReg[14] |= 0x20;
+                break;
+            case 3:
+                gSaveContext.save.weekEventReg[14] |= 0x40;
+                break;
+            default:
+                break;
+        }
+    } else {
+        func_800B85E0(&this->actor, play, 200.0f, -1);
+    }
+}
+
+void func_80AEB828(EnLiftNuts* this) {
+    if (!func_80AE9B4C(0, 4)) {
+        func_80AE9B4C(1, 4);
+        this->unk_356 = 1;
+    }
+    if (this->actionFunc == func_80AE9F70) {
+        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 15);
+    }
+    this->unk_354 = 0;
+    this->actionFunc = func_80AEB8A4;
+}
+
+void func_80AEB8A4(EnLiftNuts* this, PlayState* play) {
+    if (this->unk_354 == 22) {
+        if (this->unk_356 == 1) {
+            Message_StartTextbox(play, 0x27F6, &this->actor);
+            this->textId = 0x27F6;
+        }
+        Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, 17);
+        this->actionFunc = func_80AEB934;
+    }
+    this->unk_354++;
+}
+
+void func_80AEB934(EnLiftNuts* this, PlayState* play) {
+    if (Animation_OnFrame(&this->skelAnime, 5.0f)) {
+        func_80AEB9E0(this, play);
+    }
+}
+
+void func_80AEB974(EnLiftNuts* this) {
+    s16 textId = this->textId;
+
+    switch (textId) {
+        default:
+            this->unk_1E4 = 0;
+            break;
```

Comment:
Does the `default` have to be at the start? Sometimes you can move it to the end

Reply:
it does